components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/WebExtensionToolbarAction.kt

```diff
@@ -64,4 +61,14 @@ open class WebExtensionToolbarAction(
         textView.setTextColor(badgeTextColor)
         textView.setBackgroundColor(badgeBackgroundColor)
     }
+
+    override fun equals(other: Any?): Boolean {
+        if (other is WebExtensionToolbarAction) {
+            return this.browserAction === other.browserAction
```

Comment:
I think comparing using reference could cause duplication maybe we could use an approach as the one suggested above where we don't have to override `equals` nor `hashCode`. What do you think?

Reply:
What other use cases for `WebExtensionToolbarAction` that would deter from overriding object value comparables vs. references?

Reply:
Normally it is not a good idea to override `equals` and `hashcode` unless is completely necessarily. Because after we do it, we have a maintenance cost, as we have to watch for these functions after every change in the class, for this is better to use [data classes](https://kotlinlang.org/docs/reference/data-classes.html) that save of this maintenance work, as they regenerate `equals()` and `hashCode()` for use.

`WebExtensionToolbarAction` is not **data class** now because, `data classes` doesn't allow inheritance :( and as **browser actions** and **page actions** look exactly the same the only differences is the place where they are located, I was thinking about reusing the behavior on `WebExtensionToolbarAction` for page action because we only really need another type with the same functionality.

**pageAction** Inside of the url
![image](https://user-images.githubusercontent.com/773158/68217991-9c6a0980-ffb1-11e9-968d-e5f38f22291b.png)

**browserAction** Inside of the toolbar
![image](https://user-images.githubusercontent.com/773158/68218056-b6a3e780-ffb1-11e9-877d-aedd60434b52.png)